### PR TITLE
Boost: add docs for filters in the Cache module

### PR DESCRIPTION
**Pull request by donnchawp**
Add documentation to Cache_API.md for the filters used in this part of the plugin.

Also see #36209

## Proposed changes:
* add docs to Cache_API.md

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:
No testing possible. Just documentation

### CODE REVIEW

File: projects/plugins/boost/docs/Cache_API.md [hunk 1]
```diff
@@ -51,3 +51,38 @@ Usage:
...
+
+### Modify the cache bypass filters
+
+Filter hook: `boost_cache_bypass_patterns`
+
+Parameter: an array of regex patterns that define URLs that bypass caching
+
+Usage:
+```php
+apply_filters( 'boost_cache_bypass_patterns', array( '/shop', '/checkout', '/200?/(.*)' ) );
```
Comment (dilirity) [inline projects/plugins/boost/docs/Cache_API.md]: The names of the filters changed when we merged https://github.com/Automattic/jetpack/pull/36209.

File: projects/plugins/boost/docs/Cache_API.md [hunk 2]
```diff
@@ -51,3 +51,38 @@ Usage:
...
+
+### Override if the current request is cacheable
+
+Filter hook: `boost_cache_cacheable`
+
+Parameter: bool
+
+Usage:
+```php
+apply_filters( 'boost_cache_cacheable', true );
```
Comment (dilirity) [inline projects/plugins/boost/docs/Cache_API.md]: Though the name changed, the second param is the request URI.

**Review by dilirity — DISMISSED** [review]

File: projects/plugins/boost/docs/Cache_API.md (hunk 1 above)
Comment (donnchawp) [reply projects/plugins/boost/docs/Cache_API.md]: Done. I had updated last week, but I forgot to ping you.

**Review by dilirity — APPROVED** [review]
This is ready for merge.

:shipit:

**Pull request merged**